python-package/lightgbm/dask.py

```diff
@@ -169,24 +180,33 @@ def _split_to_parts(data, is_matrix):
     return parts
 
 
-def _train(client, data, label, params, model_factory, sample_weight=None, group=None, **kwargs):
+def _train(
+    client: Client,
+    data: _DaskCollection,
+    label: _DaskCollection,
+    params: Dict[str, Any],
+    model_factory: Type[LGBMModel],
+    sample_weight: Optional[_DaskCollection] = None,
+    group: Optional[_DaskCollection] = None,
+    **kwargs: Any
+) -> LGBMModel:
     """Inner train routine.
 
     Parameters
     ----------
     client : dask.distributed.Client
         Dask client.
-    data : dask array of shape = [n_samples, n_features]
+    data : dask Array or dask DataFrame of shape = [n_samples, n_features]
         Input feature matrix.
-    label : dask array of shape = [n_samples]
+    label : dask Array, dask DataFrame, or dask Series of shape = [n_samples]
         The target values (class labels in classification, real numbers in regression).
     params : dict
         Parameters passed to constructor of the local underlying model.
     model_factory : lightgbm.LGBMClassifier, lightgbm.LGBMRegressor, or lightgbm.LGBMRanker class
```

Comment:
Maybe simplify?
```suggestion
    model_factory : subclass of lightgbm.LGBMModel
```
And here (I can't add suggestion for that line):
https://github.com/microsoft/LightGBM/blob/c4b51317f8f722f60105335e5c799877a7dd80bd/python-package/lightgbm/dask.py#L200

Reply:
That would be less words, but I disagree that it would be "simpler". I think listing the three classes is easier to understand than saying "subclass of LGBMModel", especially since outside contributors might not know that `LGBMClassifier` (for example) is a child of `LGBMModel` without looking it up first.

Since I'm not aware of any plan to add a new, 4th child of `LGBMModel` any time soon, I'd rather keep this as-is.

Reply:
Since this is an internal function though, I don't feel strongly. So if you want me to change it I will.

Reply:
Nope, I'm fine with your reasoning. Thanks!